docker/ubuntu/16.04/cpu_ispc_build/Dockerfile

```diff
@@ -20,30 +20,54 @@ ARG EXTRA_BUILD_ARG
-RUN apt-get -y update && apt-get --no-install-recommends install -y software-properties-common && add-apt-repository ppa:deadsnakes/ppa && \
-    apt-get -y update && apt-get --no-install-recommends install -y wget build-essential gcc g++ git python3.6 ncurses-dev libtinfo-dev libtbb-dev && \
+RUN apt-get -y update && apt-get --no-install-recommends install -y software-properties-common && \
+    apt-get -y upgrade && apt-get --no-install-recommends install -y wget build-essential git ncurses-dev libtinfo-dev libtbb-dev && \
     rm -rf /var/lib/apt/lists/*
 
-# ISPC and LLVM starting 16.0 build in C++17 mode and will fail without modern libstdc++
+# ISPC and LLVM starting 16.0 build in C++17 mode and will fail without modern libstdc++ and modern compiler (gcc>=7).
 RUN add-apt-repository ppa:ubuntu-toolchain-r/test -y && apt-get -y update && \
-    apt-get --no-install-recommends install -y libstdc++-9-dev && \
+    apt-get --no-install-recommends install -y libstdc++-9-dev gcc-7 g++-7 && \
     rm -rf /var/lib/apt/lists/*
 
+RUN update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-7 10 && \
+    update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-7 10 && \
+    update-alternatives --install /usr/bin/cc cc /usr/bin/gcc 30 && \
+    update-alternatives --set cc /usr/bin/gcc && \
+    update-alternatives --install /usr/bin/c++ c++ /usr/bin/g++ 30 && \
+    update-alternatives --set c++ /usr/bin/g++
+
 # Download and install required version of cmake (3.20) for ISPC build
-RUN wget -q --retry-connrefused --waitretry=5 --read-timeout=20 --timeout=15 -t 5 https://github.com/Kitware/CMake/releases/download/v3.20.5/cmake-3.20.5-linux-x86_64.sh && mkdir /opt/cmake && sh cmake-3.20.5-linux-x86_64.sh --prefix=/opt/cmake --skip-license && \
-    ln -s /opt/cmake/bin/cmake /usr/local/bin/cmake && rm cmake-3.20.5-linux-x86_64.sh
+RUN wget -q --retry-connrefused --waitretry=5 --read-timeout=20 --timeout=15 -t 5 https://github.com/Kitware/CMake/releases/download/v3.20.5/cmake-3.20.5-linux-x86_64.sh && \
+    sh cmake-3.20.5-linux-x86_64.sh --prefix=/usr/local --skip-license && rm cmake-3.20.5-linux-x86_64.sh
 
 # If you are behind a proxy, you need to configure git.
-RUN if [ -v "$http_proxy" ]; then git config --global --add http.proxy "$http_proxy"; fi
+RUN if [ -v http_proxy ]; then git config --global --add http.proxy "$http_proxy"; fi
 
 WORKDIR /usr/local/src
 
+# Zlib is required to build Python3.8.
+RUN apt-get -y update && \
+    apt-get -y install zlib1g-dev zlib1g && \
+    rm -rf /var/lib/apt/lists/*
+RUN wget -q --retry-connrefused --waitretry=5 --read-timeout=20 --timeout=15 -t 5 https://www.python.org/ftp/python/3.8.18/Python-3.8.18.tgz && \
+    tar xf Python-3.8.18.tgz && cd Pyth* && \
+    ./configure && make -j"$(nproc)" && make install && \
+    rm -rf /usr/local/src/*
+
+# git >= 2.7.0 is required to use --shallow-submodules.
+RUN apt-get -y update && \
```

Comment:
Let's not use `--shallow-submodules` in this Dockerfile. It introduces installation of extra dependencies.

Reply:
I deleted git and adjusted calls to `alloy.py`